### PR TITLE
Issue #5759: Update AbstractChecks to log DetailAST (part 1)

### PR DESCRIPTION
**Pull request by pbludov**
Issue #5759 

Regression tests:
https://pbludov.github.io/issue-5759/

Note that patch version reports a bit more violations, for example:
https://pbludov.github.io/issue-5759/checkstyle/index.html#A1151

### CODE REVIEW

**Comment by rnveach** [comment]
@romani Do we require xpath tests for every check in this issue?

**Comment by romani** [comment]
Yes, we demanded this before, we need to keep doing it, as it is the only reason to have such change.

**Comment by rnveach** [comment]
@pbludov See https://github.com/checkstyle/checkstyle/issues/6207 for what tests we are asking for for the checks changing in this PR.

**Comment by pbludov** [comment]
> @romani Do we require xpath tests for every check in this issue?

done.

File: src/it/java/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/GenericWhitespaceTest.java [hunk 1]
```diff
@@ -42,22 +42,22 @@ public void testWhitespaceAroundGenerics() throws Exception {
...
-            "14:32: " + getCheckMessage(messages, msgPreceded, "<"),
-            "14:34: " + getCheckMessage(messages, msgFollowed, "<"),
-            "14:45: " + getCheckMessage(messages, msgPreceded, ">"),
-            "15:32: " + getCheckMessage(messages, msgPreceded, "<"),
-            "15:34: " + getCheckMessage(messages, msgFollowed, "<"),
-            "15:45: " + getCheckMessage(messages, msgPreceded, ">"),
-            "20:38: " + getCheckMessage(messages, msgPreceded, "<"),
-            "20:40: " + getCheckMessage(messages, msgFollowed, "<"),
-            "20:61: " + getCheckMessage(messages, msgPreceded, ">"),
+            "12:17: " + getCheckMessage(messages, msgFollowed, "<"),
```
Comment (romani) [inline src/it/java/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/GenericWhitespaceTest.java]: +1 is correct, verified at InputWhitespaceAroundGenerics, symbol `<` is at 17 column.

**Comment by romani** [comment]
@pbludov , verification of such big update is very complicated.
do you have time to split it in several parts ?

**Pull request closed without merging**